### PR TITLE
implements left outer join support to NL operation

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

**Comment by seut** [comment]
@matriv @dobe  pls review

File: sql/src/main/java/io/crate/operation/join/NestedLoopOperation.java [hunk 1]
```diff
@@ -382,19 +392,37 @@ private Result emitRow(Row row) {
...
+            Result result = emitRowAndTrace(combinedRow);
             if (result == Result.PAUSE) {
                 pauseFromDownstream = true;
             }
             return result;
         }
 
+        private RowReceiver.Result emitRowAndTrace(Row row) {
+            RowReceiver.Result result = downstream.setNextRow(row);
+            if (LOGGER.isTraceEnabled() && result != RowReceiver.Result.CONTINUE) {
```
Comment (matriv) [inline sql/src/main/java/io/crate/operation/join/NestedLoopOperation.java]: You can remove LOGGER.isTraceEnabled(), no expensive object evaluation.

File: sql/src/test/java/io/crate/operation/join/NestedLoopOperationTest.java [hunk 2]
```diff
@@ -308,6 +319,26 @@ public void run() {
...
+
+        Predicate<Row> rowFilterPredicate = new Predicate<Row>() {
+            @Override
+            public boolean apply(@Nullable Row input) {
+                return input.get(0) == input.get(1);
+            }
+        };
+        Bucket rows = executeNestedLoop(leftRows, rightRows, rowFilterPredicate, JoinType.LEFT);
+
+        assertThat(TestingHelpers.printedTable(rows), is("" +
```
Comment (matriv) [inline sql/src/test/java/io/crate/operation/join/NestedLoopOperationTest.java]: Remove the empty string ""

**Comment by matriv** [comment]
Through

**Comment by matriv** [comment]
Igtm

**Pull request merged**